### ISSUE TITLE
Problems running Migrations Script command

### ISSUE THREAD

**Issue by dneimke** [dotnet/efcore#5977]
### Background

Running the following command emits a list of Migrations in my project

`> dotnet ef migrations list`

Output:
- 20160705022944_AAA
- 20160705023838_BBB

The migrations history table in the database looks like this:

| MigrationId | ProductVersion |
| --- | :-: |
| 20160705022944_AAA | 1.0.0-rc2-20896 |

Running the following command emits a sql script which contains the content of both migrations and creates insert statements for both migration history items:

`> dotnet ef migrations script -o deploy.sql`

Output:

```
  snip...
 INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
 VALUES (N'20160705022944_AAA', N'1.0.0-rc2-20896');

  snip...
 INSERT INTO [__EFMigrationsHistory] ([MigrationId], [ProductVersion])
 VALUES (N'20160705022944_BBB', N'1.0.0-rc2-20896');

```
### The issue

Running the following command emits a sql script which contains the drop step for 20160705023838_BBB:

`> dotnet ef migrations script from AAA -o deploy.sql`

**EXPECTED:** A sql script containing the UP statements for BBB
**ACTUAL:** A sql script containing the DOWN statements for BBB

**QUESTION:** Why does the content for the Drop step get output here?

Running the following command produces an error message:

> dotnet ef migrations script from AAA to BBB -o deploy.sql
> **"Unrecognized command or argument 'to'"**

**QUESTION:** Is that the correct syntax for specifying a [to]?
### Further technical details

EF Core version: 1.0.0-rc2-final

Dependencies:

```
"Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview1-final",
      "type": "build"
    },
```

Tools/Frameworks

```
"tools": {
    "Microsoft.EntityFrameworkCore.Tools": {
      "version": "1.0.0-preview1-final",
      "imports": [
        "portable-net45+win8+dnxcore50",
        "portable-net45+win8"
      ]
    }
  },
  "frameworks": {
    "net451": {
      "frameworkAssemblies": {
        "System.ComponentModel": "4.0.0.0"
      }
    }    
  }
```

Operating system: 
Visual Studio version: N/A

Other details about my project setup:

**Comment by lnaie** [comment]
This is your command actually: dotnet ef migrations script AAA -o deploy.sql

**Comment by dneimke** [comment]
Thanks @lnaie.  I also noticed that we needed to put in the full key of the migration.  E.g. 

dotnet ef migrations script 20160705022944_AAA -o deploy.sql

This must have changed at some stage?

**Comment by cspaniard** [comment]
Hi, I still have to specify the full key migration id vs the simple name.

I would assume just the simplified name (the one provided upon creation) should be enough. Looking at the docs and --help I would not guess the full key was required.

My guess this is an issue to be fixed, then again I could be wrong ;)

Version: 1.0.0-preview2-21431

**Comment by bricelam** [comment]
Not being able to use the short name is issue #6228 which will be fixed in 1.0.1
